### PR TITLE
Tapir swagger scala3

### PR DESCRIPTION
**Pull request by KapStorm**
Implemented swagger using Tapir

image

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -215,12 +213,12 @@ lazy val playSettings: Project => Project = {
       libraryDependencies ++= Seq(
         "org.scalatestplus.play" %% "scalatestplus-play" % "6.0.0-M4" % Test,
         "org.scalatestplus" %% "mockito-4-6" % "3.2.15.0" % Test
-
       )
     )
 }
 
 lazy val common = (crossProject(JSPlatform, JVMPlatform) in file("lib/common"))
+  .dependsOn(tapirPlayJson)
```
Comment (AlexITC) [inline build.sbt:221]: Is this really required?

File: build.sbt [hunk 2]
```diff
@@ -309,8 +306,38 @@ lazy val ui = (project in file("lib/ui"))
...
+  )
+
+lazy val tapirServerPlay = (project in file("tapir/tapir-play"))
+  .settings(
+    name := "tapir-server-play",
+    scalaVersion := "3.3.0-RC3",
+    libraryDependencies ++= Seq(
+      "com.typesafe.play" %% "play-akka-http-server" % "2.9.0-M5",
+      "com.softwaremill.sttp.shared" %% "akka" % "1.3.14",
+      "org.scala-lang.modules" %% "scala-collection-compat" % "2.11.0"
```
Comment (AlexITC) [inline build.sbt:322]: Is this necessary?

File: build.sbt [hunk 3]
```diff
@@ -309,8 +306,38 @@ lazy val ui = (project in file("lib/ui"))
...
+      "com.softwaremill.sttp.tapir" %% "tapir-core" % tapir
+    )
+  )
+  .jsSettings(
+    libraryDependencies += "io.github.cquiroz" %%% "scala-java-time" % scalaJavaTime
+  )
+
 lazy val server = (project in file("server"))
-  .dependsOn(common.jvm, api.jvm)
+  .dependsOn(common.jvm, api.jvm, tapirServerPlay, tapirPlayJson.js)
```
Comment (AlexITC) [inline build.sbt:340]: Must be `tapirPlayJson.jvm`

File: lib/api/shared/src/main/scala/net/wiringbits/api/models/AdminGetUserLogs.scala [hunk 4]
```diff
@@ -1,18 +1,22 @@
...
 
 import java.time.Instant
 import java.util.UUID
 
 object AdminGetUserLogs {
-  @ApiModel(value = "AdminGetUserLogsResponse", description = "Includes the logs for a single user")
   case class Response(data: List[Response.UserLog])
   implicit val adminGetUserLogsResponseFormat: Format[Response] = Json.format[Response]
+  implicit val adminGetUserLogsResponseSchema: Schema[Response] =
+    Schema.derived[Response].name(Schema.SName("AdminGetUserLogsResponse"))
```
Comment (AlexITC) [inline lib/api/shared/src/main/scala/net/wiringbits/api/models/AdminGetUserLogs.scala:14]: What about the description?

File: server/src/main/resources/routes [hunk 5]
```diff
@@ -41,8 +41,7 @@ GET /admin/users/:userId/logs controllers.AdminController.getUserLogs(userId)
 # get frontend configs
 GET /environment-config controllers.EnvironmentConfigController.getEnvironmentConfig()
 
-# static assets for swagger-ui
-GET /docs/*file        controllers.Assets.at(path="/swagger-ui", file)
+-> / controllers.ApiRoutes
```
Comment (AlexITC) [inline server/src/main/resources/routes:44]: Existing routes must be server through swagger

File: server/src/main/scala/controllers/AdminController.scala [hunk 6]
```diff
@@ -65,3 +39,62 @@ class AdminController @Inject() (
...
   }
 }
+
+object AdminController {
+  import sttp.model.StatusCode
+  import sttp.tapir.*
+  import sttp.tapir.json.play.*
+
+  private val getUserLogsEndpoint = endpoint.get
+    .in("admin" / "users" / path[UUID]("userId") / "logs")
```
Comment (AlexITC) [inline server/src/main/scala/controllers/AdminController.scala:49]: This is repeating what we have in the routes file unnecessarily

**Review by AlexITC — CHANGES_REQUESTED** [review]
Let's split this PR in two:
1. To add the tapir modules
2. To add do the switch to tapir.


Once we get everything working, we'll need to raise a PR to Tapir.

File: server/src/main/resources/routes (hunk 5 above)
Comment (KapStorm) [reply server/src/main/resources/routes:44]: Added a TODO for next PR

File: server/src/main/scala/controllers/AdminController.scala (hunk 6 above)
Comment (KapStorm) [reply server/src/main/scala/controllers/AdminController.scala:49]: Added a TODO for next PR

**Pull request merged**